packages/vite/src/node/ssr/fetchModule.ts

```diff
@@ -148,13 +139,10 @@ function inlineSourceMap(
   if (OTHER_SOURCE_MAP_REGEXP.test(code))
     code = code.replace(OTHER_SOURCE_MAP_REGEXP, '')
 
-  const sourceMap = Buffer.from(
-    JSON.stringify(processSourceMap?.(map) || map),
-    'utf-8',
-  ).toString('base64')
+  const sourceMap = processSourceMap?.(map) || map
   result.code = `${code.trimEnd()}\n//# sourceURL=${
     mod.id
-  }\n${VITE_RUNTIME_SOURCEMAPPING_SOURCE}\n//# ${VITE_RUNTIME_SOURCEMAPPING_URL};base64,${sourceMap}\n`
```

Comment:
I removed `charset=utf-8` part.
[JSON should be always utf-8](https://datatracker.ietf.org/doc/html/rfc8259#:~:text=JSON%20text%20exchanged%20between%20systems%20that%20are%20not%20part%20of%20a%20closed%0A%20%20%20ecosystem%20MUST%20be%20encoded%20using%20UTF%2D8) and [the charset in mime doesn't have any effect](https://datatracker.ietf.org/doc/html/rfc8259#:~:text=Note%3A%20%20No%20%22charset%22%20parameter%20is%20defined%20for%20this%20registration.%0A%20%20%20%20%20%20Adding%20one%20really%20has%20no%20effect%20on%20compliant%20recipients.) according to the RFC.